### PR TITLE
Extract CRM Bundle

### PR DESCRIPTION
**Pull request by wachterjohannes**
__Tasks:__

- [ ] gather feedback for my changes
- [x] added changelog line

__Informations:__

| Q             | A
| ------------- | ---
| Tests pass?   | yes
| Fixed tickets | TODO
| BC Breaks     | none
| Doc           | none

### CODE REVIEW

File: src/Sulu/Bundle/AdminBundle/Controller/WidgetGroupsController.php [hunk 1]
```diff
@@ -36,14 +36,14 @@ public function groupAction($groupAlias, Request $request)
...
-                )
+            $content = $this->getWidgetsHandler()->renderWidgetGroup(
+                $groupAlias,
+                $request->query->all()
             );
+
+            return new Response($content, $content !== '' ? 200 : 204);
         } catch (WidgetException $ex) {
-            return new Response($ex->getMessage());
+            return new Response($ex->getMessage(), 404);
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Controller/WidgetGroupsController.php]: Is 404 really the correct error code here? Is the only error that can appear that the widget doesn't exist? Then I would rename the exception to `WidgetNotFoundException` or even create a new sub exception.

File: src/Sulu/Bundle/AdminBundle/Navigation/ContentNavigation.php [hunk 2]
```diff
@@ -71,16 +71,27 @@ public function toArray($contentType = null)
...
+                    $navigationItem->getPosition() >= count($navigationItems)
+                ) {
+                    $navigationItems[] = $navigationItem->toArray();
+                } else {
+                    array_splice(
+                        $navigationItems,
+                        $navigationItem->getPosition() - 1,
+                        0,
+                        array($navigationItem->toArray())
+                    );
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Navigation/ContentNavigation.php]: What should this statement do?

File: src/Sulu/Bundle/AdminBundle/Resources/public/js/components/sidebar/main.js [hunk 3]
```diff
@@ -344,6 +344,11 @@ define([], function() {
                     var $widget;
                     this.emptySidebar(false);
                     this.loadWidget(url).then(function(widget) {
+                        if (widget === undefined || widget.length === '') {
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/public/js/components/sidebar/main.js]: What should the `widget.length` check prevent?

File: src/Sulu/Bundle/AdminBundle/Resources/public/js/components/sidebar/widget-groups.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ *
+ */
+
+/**
+ * @class WidgetGroups
+ * @constructor
+ */
+
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/public/js/components/sidebar/widget-groups.js]: no new line

**Comment by danrot** [comment]
Can you also link the corresponding husky PR?

File: src/Sulu/Bundle/AdminBundle/Widgets/WidgetsHandler.php [hunk 5]
```diff
@@ -88,10 +96,18 @@ public function render($aliases, $parameters = array())
...
+                $this->template,
+                array(
+                    'widgets' => $widgets,
+                    'parameters' => $parameters
+
+                )
+            );
+        }
+
+        return '';
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Widgets/WidgetsHandler.php]: Why not returning `null`?

File: src/Sulu/Bundle/CategoryBundle/Controller/CategoryController.php [hunk 6]
```diff
@@ -318,6 +325,29 @@ protected function getCategoryListRepresentation(Request $request)
     }
 
     /**
+     * append parent selector to listbuilder
+     * @param $parentKey
+     * @param DoctrineListBuilder $listBuilder
+     */
+    protected function appendParent($parentKey, DoctrineListBuilder $listBuilder)
```
Comment (danrot) [inline src/Sulu/Bundle/CategoryBundle/Controller/CategoryController.php]: `append` suggests that you can add multiple parents. Is this really possible?

File: src/Sulu/Bundle/ContactBundle/Api/Account.php [hunk 7]
```diff
@@ -1166,4 +1086,24 @@ public function getMedias()
...
+
+    /**
+     * Get categories
+     *
+     * @return Category[]
+     * @VirtualProperty
+     * @SerializedName("categories")
+     * @Groups({"fullAccount"})
+     */
+    public function getCategories()
```
Comment (danrot) [inline src/Sulu/Bundle/ContactBundle/Api/Account.php]: Don't you need a setter?

File: src/Sulu/Bundle/ContactBundle/Contact/AbstractContactManager.php [hunk 8]
```diff
@@ -456,7 +467,7 @@ private function getAddresses($entity)
      */
     public function getAddressByCondition($entity, callable $conditionCallback, $force = false)
     {
-        
+
```
Comment (danrot) [inline src/Sulu/Bundle/ContactBundle/Contact/AbstractContactManager.php]: no new line

File: src/Sulu/Bundle/ContactBundle/Contact/AccountFactory.php [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+/**
+ * factory to encapsulate account creation
+ */
+class AccountFactory
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function create()
```
Comment (danrot) [inline src/Sulu/Bundle/ContactBundle/Contact/AccountFactory.php]: Maybe call this one `createEntity`? Just to be more explicit.

File: src/Sulu/Bundle/ContactBundle/Contact/AbstractContactManager.php [hunk 10]
```diff
@@ -38,11 +37,18 @@
...
+     * @var string
+     */
+    protected $accountEntity;
+
+    /**
      * @param ObjectManager $em
+     * @param $accountEntityName
      */
-    public function __construct(ObjectManager $em)
+    public function __construct(ObjectManager $em, $accountEntityName)
```
Comment (danrot) [inline src/Sulu/Bundle/ContactBundle/Contact/AbstractContactManager.php]: Is the `accountEntityName` really needed here? It's not used in this class, and set in the sub classes again.

File: src/Sulu/Bundle/ContactBundle/Contact/AccountFactoryInterface.php [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\ContactBundle\Contact;
+
+use Sulu\Bundle\ContactBundle\Api\Account;
+use Sulu\Bundle\ContactBundle\Entity\AccountInterface;
+
+interface AccountFactoryInterface
```
Comment (danrot) [inline src/Sulu/Bundle/ContactBundle/Contact/AccountFactoryInterface.php]: phpdoc missing

File: src/Sulu/Bundle/ContactBundle/Contact/AccountManager.php [hunk 12]
```diff
@@ -13,30 +13,35 @@
...
+     */
+    private $accountFactory;
+
+    public function __construct(
+        ObjectManager $em,
+        TagmanagerInterface $tagManager,
+        AccountFactory $accountFactory,
+        $accountEntityName
+    ) {
+        parent::__construct($em, $accountEntityName);
```
Comment (danrot) [inline src/Sulu/Bundle/ContactBundle/Contact/AccountManager.php]: Ok, you set it here... But again, is this really needed?

File: src/Sulu/Bundle/ContactBundle/Contact/ContactManager.php [hunk 13]
```diff
@@ -159,13 +156,12 @@ public function setMainAccount(ContactEntity $contact, $data)
         ) {
             $accountId = $data['account']['id'];
 
-            /** @var Account $account */
             $account = $this->em
-                ->getRepository(self::$accountEntityName)
+                ->getRepository($this->accountEntity)
```
Comment (danrot) [inline src/Sulu/Bundle/ContactBundle/Contact/ContactManager.php]: Ok, finally I see it on my own :)

File: src/Sulu/Bundle/ContactBundle/Controller/AbstractContactController.php [hunk 14]
```diff
@@ -132,6 +131,11 @@ protected function addNewContactRelations($contact, Request $request)
                 $this->addTag($contact, $tag);
             }
         }
+
+        // process details
+        if (!is_null($request->get('bankAccounts'))) {
```
Comment (danrot) [inline src/Sulu/Bundle/ContactBundle/Controller/AbstractContactController.php]: Please don't start using `is_null`, we almost (?) always used `=== null` until now, which is also a little bit faster.

File: src/Sulu/Bundle/ContactBundle/Controller/AccountController.php [hunk 15]
```diff
@@ -82,12 +73,13 @@ class AccountController extends AbstractContactController
     protected $accountContactFieldDescriptors;
     protected $accountAddressesFieldDescriptors;
 
-    public function __construct()
+    public function setContainer(ContainerInterface $container = null)
```
Comment (danrot) [inline src/Sulu/Bundle/ContactBundle/Controller/AccountController.php]: Why did you change this to `setContainer` instead of using the constructor? Adding these methods is not what this method is thought for.

File: src/Sulu/Bundle/ContactBundle/Controller/AccountController.php [hunk 16]
```diff
@@ -620,21 +579,61 @@ public function putAction($id, Request $request)
...
+
+        // set disabled
+        $disabled = $request->get('disabled');
+        if (!is_null($disabled)) {
+            $account->setDisabled($disabled);
+        }
+
+        if (!is_null($request->get('uid'))) {
+            $account->setUid($request->get('uid'));
+        }
```
Comment (danrot) [inline src/Sulu/Bundle/ContactBundle/Controller/AccountController.php]: Same for these two checks, please use `=== null`.

File: src/Sulu/Bundle/ContactBundle/Controller/ContactController.php [hunk 17]
```diff
@@ -79,8 +80,14 @@ class ContactController extends AbstractContactController
 
     protected $accountContactFieldDescriptors;
 
-    // TODO: move field descriptors to a manager
-    public function __construct()
+    public function setContainer(ContainerInterface $container = null)
```
Comment (danrot) [inline src/Sulu/Bundle/ContactBundle/Controller/ContactController.php]: Same as above, why misusing this method?

File: src/Sulu/Bundle/ContactBundle/Entity/BankAccount.php [hunk 18]
```diff
@@ -182,4 +182,42 @@ public function getBankName()
     {
         return $this->bankName;
     }
+    /**
+     * @var \Doctrine\Common\Collections\Collection
+     */
+    private $contacts;
```
Comment (danrot) [inline src/Sulu/Bundle/ContactBundle/Entity/BankAccount.php]: Put variables at top, and only one new line.

File: src/Sulu/Bundle/ContactBundle/Entity/BaseAccount.php [hunk 19]
```diff
@@ -0,0 +1,521 @@
...
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\ContactBundle\Entity;
+
+use JMS\Serializer\Annotation\Exclude;
+use Sulu\Bundle\CoreBundle\Entity\ApiEntity;
+use Sulu\Component\Persistence\Model\AuditableInterface;
+
+abstract class BaseAccount extends ApiEntity implements AuditableInterface, AccountInterface
```
Comment (danrot) [inline src/Sulu/Bundle/ContactBundle/Entity/BaseAccount.php]: phpdoc missing

File: src/Sulu/Bundle/ContactBundle/Entity/Contact.php [hunk 20]
```diff
@@ -1226,4 +1147,88 @@ public function getCategories()
     {
         return $this->categories;
     }
+
+    /**
+     * @var \Doctrine\Common\Collections\Collection
+     */
+    private $bankAccounts;
```
Comment (danrot) [inline src/Sulu/Bundle/ContactBundle/Entity/Contact.php]: put variables at the top

**Comment by danrot** [comment]
Please also remove `TermsOfDeliveryController`, `TermsOfPaymentController` and the financial tab in the UI.

File: src/Sulu/Bundle/ContactBundle/Resources/public/dist/input-type/iban-input.js [hunk 21]
```diff
@@ -0,0 +1 @@
+define(["type/default"],function(a){"use strict";return function(b,c){var d={id:"id",label:"value",required:!1},e={setValue:function(a){this.$el.data({value:app.sandbox.iban.printFormat(a)}).trigger("data-changed")},getValue:function(){return this.$el.find("input").val()},needsValidation:function(){return""!==this.getValue()},validate:function(){var a=/^([a-zA-Z0-9 ])+$/;return app.sandbox.iban.isValid(this.getValue())&&a.test(this.getValue())}};return new a(b,d,c,"iban-input",e)}});
```
Comment (danrot) [inline src/Sulu/Bundle/ContactBundle/Resources/public/dist/input-type/iban-input.js]: Shouldn't this file be removed?

**Comment by danrot** [comment]
Still stumbling upon some iban components, shouldn't all of them be removed?

File: src/Sulu/Bundle/ContactBundle/Resources/public/js/components/accounts/components/form/main.js [hunk 22]
```diff
@@ -19,18 +19,35 @@ define(['config'], function(Config) {
...
+            bankAccountAddSelector: '.bank-account-add'
+        },
+
+        customTemplates = {
+            addBankAccountsIcon: [
+                '<div class="grid-row">',
+                '    <div class="grid-col-12">',
+                '       <span id="bank-account-add" class="fa-plus-circle icon bank-account-add clickable pointer m-left-140"></span>',
+                '   </div>',
+                '</div>'].join('')
```
Comment (danrot) [inline src/Sulu/Bundle/ContactBundle/Resources/public/js/components/accounts/components/form/main.js]: Put `].join('')` into the next line.

File: src/Sulu/Bundle/ContactBundle/Resources/public/js/components/accounts/components/form/main.js [hunk 23]
```diff
@@ -103,12 +109,20 @@ define(['config'], function(Config) {
...
+                    },
+                    {
+                        name: 'input@husky',
+                        options: {
+                            el: '#vat',
+                            instanceName: 'vat-input',
+                            value: !!data.uid ? data.uid : ''
+                        }
+                    }
+                ]);
```
Comment (danrot) [inline src/Sulu/Bundle/ContactBundle/Resources/public/js/components/accounts/components/form/main.js]: wrong indentation

File: src/Sulu/Bundle/ContactBundle/Resources/public/js/components/accounts/components/form/main.js [hunk 24]
```diff
@@ -410,11 +289,19 @@ define(['config'], function(Config) {
...
         setFormData: function(data) {
             // add collection filters to form
             this.sandbox.emit('sulu.contact-form.add-collectionfilters', this.form);
+
+            this.numberOfBankAccounts = !!data.bankAccounts ? data.bankAccounts.length : 0;
+            this.updateBankAccountAddIcon(this.numberOfBankAccounts);
+
             this.sandbox.form.setData(this.form, data).then(function() {
-                this.sandbox.start(this.form);
+                this.sandbox.start('#contact-fields');
```
Comment (danrot) [inline src/Sulu/Bundle/ContactBundle/Resources/public/js/components/accounts/components/form/main.js]: Is there a reason for this change? At least I would put that string into the `constants`-variable.

File: src/Sulu/Bundle/ContactBundle/Resources/public/js/components/accounts/components/list/main.js [hunk 25]
```diff
@@ -9,20 +9,13 @@
...
     'use strict';
 
     var bindCustomEvents = function() {
-            // navigate to edit account
-            this.sandbox.on('husky.datagrid.item.click', function(id) {
-                this.sandbox.emit(
-                    'sulu.sidebar.set-widget',
-                    '/admin/widget-groups/account-info?account=' + id
-                );
-            }, this);
```
Comment (danrot) [inline src/Sulu/Bundle/ContactBundle/Resources/public/js/components/accounts/components/list/main.js]: If you remove these lines, you probably also don't need the WidgetGroups, right?

File: src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php [hunk 26]
```diff
@@ -240,6 +245,26 @@ protected function addIns(array $inFields, array $inValues)
...
+            $betweenParts[] = $betweenField->getSelect() .
+                ' BETWEEN :' . $betweenField->getName() . '1'.
+                ' AND :' . $betweenField->getName() . '2';
+            $values = $betweenValues[$betweenField->getName()];
+            $this->queryBuilder->setParameter($betweenField->getName() . '1', $values[0]);
+            $this->queryBuilder->setParameter($betweenField->getName() . '2', $values[1]);
+        }
+
+        $this->queryBuilder->andWhere('(' . implode(' AND ', $betweenParts) . ')');
+    }
```
Comment (danrot) [inline src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php]: Is it possible that tests are missing for this functionality?

**Comment by danrot** [comment]
I would add the resolve target entity for the tests in the TestBundle, where it is valid for all tests, instead of adding it to every single bundle.

File: src/Sulu/Bundle/AdminBundle/Controller/WidgetGroupsController.php (hunk 1 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/AdminBundle/Controller/WidgetGroupsController.php]: no i will check before and return 404 if widget group does not exists otherwise 400!

File: src/Sulu/Bundle/AdminBundle/Navigation/ContentNavigation.php (hunk 2 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/AdminBundle/Navigation/ContentNavigation.php]: add item at given position!

**Comment by wachterjohannes** [comment]
corresponding husky PR https://github.com/massiveart/husky/pull/466

File: src/Sulu/Bundle/AdminBundle/Resources/public/js/components/sidebar/main.js (hunk 3 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/AdminBundle/Resources/public/js/components/sidebar/main.js]: it should only be `widget !== ''` i will fix it

File: src/Sulu/Bundle/AdminBundle/Widgets/WidgetsHandler.php (hunk 5 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/AdminBundle/Widgets/WidgetsHandler.php]: fixed

File: src/Sulu/Bundle/CategoryBundle/Controller/CategoryController.php (hunk 6 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/CategoryBundle/Controller/CategoryController.php]: renamed to `addParentSelector`

File: src/Sulu/Bundle/ContactBundle/Api/Account.php (hunk 7 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContactBundle/Api/Account.php]: no not needed ... set categories is done by the manager which uses the entitites

File: src/Sulu/Bundle/ContactBundle/Contact/AbstractContactManager.php (hunk 8 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContactBundle/Contact/AbstractContactManager.php]: fixed

File: src/Sulu/Bundle/ContactBundle/Contact/AccountFactory.php (hunk 9 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContactBundle/Contact/AccountFactory.php]: fixed

File: src/Sulu/Bundle/ContactBundle/Contact/AbstractContactManager.php (hunk 10 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContactBundle/Contact/AbstractContactManager.php]: it is only forwarded in the constructor

File: src/Sulu/Bundle/ContactBundle/Contact/AccountFactoryInterface.php (hunk 11 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContactBundle/Contact/AccountFactoryInterface.php]: fixed

File: src/Sulu/Bundle/ContactBundle/Contact/AccountManager.php (hunk 12 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContactBundle/Contact/AccountManager.php]: answered in the other comment

File: src/Sulu/Bundle/ContactBundle/Contact/ContactManager.php (hunk 13 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContactBundle/Contact/ContactManager.php]: i have to wait only long enough (=

File: src/Sulu/Bundle/ContactBundle/Controller/AbstractContactController.php (hunk 14 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContactBundle/Controller/AbstractContactController.php]: fixed

File: src/Sulu/Bundle/ContactBundle/Controller/AccountController.php (hunk 15 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContactBundle/Controller/AccountController.php]: because it needs a set container for `getParameter`. and in the constructor the container is not set!

Comment (wachterjohannes) [inline src/Sulu/Bundle/ContactBundle/Controller/AccountController.php]: but i will change this to lazy loading!

Comment (wachterjohannes) [inline src/Sulu/Bundle/ContactBundle/Controller/AccountController.php]: hmm thats bad it is needed at very much places!

File: src/Sulu/Bundle/ContactBundle/Controller/AccountController.php (hunk 16 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContactBundle/Controller/AccountController.php]: fixed

File: src/Sulu/Bundle/ContactBundle/Resources/public/dist/input-type/iban-input.js (hunk 21 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContactBundle/Resources/public/dist/input-type/iban-input.js]: no it is moved to `SuluContactBundle`

File: src/Sulu/Bundle/ContactBundle/Entity/BankAccount.php (hunk 18 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContactBundle/Entity/BankAccount.php]: fixed

File: src/Sulu/Bundle/ContactBundle/Entity/Contact.php (hunk 20 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContactBundle/Entity/Contact.php]: fixed

File: src/Sulu/Bundle/ContactBundle/Resources/public/js/components/accounts/components/form/main.js (hunk 22 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContactBundle/Resources/public/js/components/accounts/components/form/main.js]: fixed

File: src/Sulu/Bundle/ContactBundle/Resources/public/js/components/accounts/components/form/main.js (hunk 24 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContactBundle/Resources/public/js/components/accounts/components/form/main.js]: yeah because i only want to start a part of the form the rest will be started otherwise!

File: src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php (hunk 26 above)
Comment (wachterjohannes) [inline src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php]: yes (=

Comment (wachterjohannes) [inline src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php]: fixed

**Comment by wachterjohannes** [comment]
@danrot i will create a new PR for move finacials tab

File: src/Sulu/Bundle/AdminBundle/Widgets/WidgetsHandler.php [hunk 27]
```diff
@@ -88,10 +96,18 @@ public function render($aliases, $parameters = array())
...
-            'parameters' => $parameters
-        ));
+        if (count($widgets) > 0) {
+            // render template
+            return $this->templateEngine->render(
+                $this->template,
+                array(
+                    'widgets' => $widgets,
+                    'parameters' => $parameters
+
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Widgets/WidgetsHandler.php]: sry, have overseen that one: no new line

File: src/Sulu/Bundle/CategoryBundle/Controller/CategoryController.php [hunk 28]
```diff
@@ -293,6 +296,10 @@ protected function getCategoryListRepresentation(Request $request)
 
         $listBuilder->addGroupBy($this->getManager()->getFieldDescriptor('id'));
 
+        if ($parentKey !== null) {
+            $this->addParentSelector($parentKey, $listBuilder);
```
Comment (danrot) [inline src/Sulu/Bundle/CategoryBundle/Controller/CategoryController.php]: `add` suggests the same... would call it `setParentKey` or something like this.

File: src/Sulu/Bundle/ContactBundle/Controller/AccountController.php (hunk 15 above)
Comment (danrot) [inline src/Sulu/Bundle/ContactBundle/Controller/AccountController.php]: Hm, that's not really a good solution either... Why not calling the `initFieldDescriptors` when it is really needed? Or replace it with a `getFieldDescriptors`-method, which lazily loads?

File: src/Sulu/Bundle/ContactBundle/Resources/public/js/components/accounts/components/form/main.js (hunk 24 above)
Comment (danrot) [inline src/Sulu/Bundle/ContactBundle/Resources/public/js/components/accounts/components/form/main.js]: Ok, but I would still put that into an own variable.

File: src/Sulu/Bundle/CategoryBundle/Controller/CategoryController.php (hunk 28 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/CategoryBundle/Controller/CategoryController.php]: but you can add multiple parents!

File: src/Sulu/Bundle/ContactBundle/Resources/public/js/components/accounts/components/form/main.js (hunk 24 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContactBundle/Resources/public/js/components/accounts/components/form/main.js]: fixed

File: src/Sulu/Bundle/AdminBundle/Widgets/WidgetsHandler.php (hunk 27 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/AdminBundle/Widgets/WidgetsHandler.php]: fixed

File: src/Sulu/Bundle/ContactBundle/Controller/ContactController.php (hunk 17 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContactBundle/Controller/ContactController.php]: fixed

File: src/Sulu/Bundle/ContactBundle/Controller/AccountController.php (hunk 15 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContactBundle/Controller/AccountController.php]: fixed

**Comment by danrot** [comment]
If I try to assign a main contact I'll get a 500 error.

**Comment by wachterjohannes** [comment]
@danrot also fixed it was a braces issue in the controller
